src/Security/Authentication/Negotiate/src/NegotiateHandler.cs

```diff
@@ -315,7 +315,7 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
             // things like ClaimsTransformation run per request.
             var identity = _negotiateState.GetIdentity();
             ClaimsPrincipal user;
-            if (identity is WindowsIdentity winIdentity)
+            if (OperatingSystem.IsWindows() && identity is WindowsIdentity winIdentity)
```

Comment:
This seems a bit odd. Are we making an extra call at runtime just to satisfy a compile-time check?

Would a valid alternative here be suppressing the platform check? Presumably on non-Windows OSes, the `identity is WindowsIdentity winIdentity` clause would work and always evaluate as `false`.

BTW I'm not worried about the extra perf cost here, since presumably `OperatingSystem.IsWindows()` is extremely cheap. Fine if you prefer to keep it. Just want to check my understanding of the situation is correct.

Reply:
Yeah, it seems like a weakness of the analyzer.

Reply:
You can use Debug.Assert(OperatingSystem.IsWindows()); instead

Reply:
After the cast? This class is fully supported on linux. The is cast here is a proxy for the platform check.

Reply:
That is an interesting scenario, we haven't covered that. cc @jeffhandley

Reply:
> The is cast here is a proxy for the platform check

I'm having a hard time imagining how this could be detected.  I was trying to think of something akin to `NotNullWhenAttribute` where an API could assert which OSPlatform we must be on in certain scenarios, but I'm coming up blank for how we could achieve that, especially in this scenario.

The two existing options are `OperatingSystem.IsWindows()` and `#pragma warning disable`.

> Presumably `OperatingSystem.IsWindows()` is extremely cheap

It's as optimized as possible; it's the equivalent of a `const` check.

Reply:
@Tratcher I'll let you decide. Should we suppress the warning and remove the runtime check?

Reply:
IsWindows is slightly cleaner than the suppression, might as well keep it.